src/store/modules/stock/mutations.ts

```diff
@@ -11,6 +11,20 @@ const mutations: MutationTree <StockState> = {
         [payload.facilityId]: payload.stock
       }
     }
+  },
+  [types.STOCK_ADD_PRODUCT_INFORMATION] (state, { productId, facilityId, minimumStock, onlineAtp, reservedQuantity }) {
+    if (!state.inventoryInformation[productId]) {
+      state.inventoryInformation[productId] = {}
```

Comment:
I think we can update the values for the stock in here itself, as if the productId does not exist in the state then there is no need to check for state on line 20.